rlw/setup_automatic_upgrade.ps1

```diff
@@ -0,0 +1,46 @@
+# This script either creates or updates a scheduled job that will run once a day that checks to
+# see if there is an upgrade for RightLink
+
+$scheduledJob = SCHTASKS.exe /Query /TN 'rightlink_check_upgrade' 2> $null
+if ($env:ENABLE_AUTO_UPGRADE -eq 'false') {
+  if ($scheduledJob) {
+    SCHTASKS.exe /Delete /TN 'rightlink_check_upgrade' /F
+    Write-Output 'Automatic upgrade disabled'
+  } else {
+    Write-Output 'Automatic upgrade never enabled - no action done'
+  }
+} else {
+  # Random hour 0-23
+  $jobHour = Get-Random -Min 0 -Max 24
+  # Put a 0 on the front of the hour for the SCHTASKS time format
+  if ($jobHour -lt 10) {
+    $jobHour = "0" + $jobHour
+  }
+
+  # Random minute 0-59
+  $jobMinute = Get-Random -Min 0 -Max 60
+  # Put a 0 on the front of the minute for the SCHTASKS time format
+  if ($jobMinute -lt 10) {
+    $jobMinute = "0" + $jobMinute
+  }
```

Comment:
This can be simplified using PowerShell's [format operator](https://technet.microsoft.com/en-us/library/hh847732(v=wps.620).aspx) `-f`:

```powershell
$jobHour = Get-Random -Min 0 -Max 24
$jobMinute = Get-Random -Min 0 -Max 60
$jobStartTime = '{0:d2}:{1:d2}' -f $jobHour, $jobMinute
```